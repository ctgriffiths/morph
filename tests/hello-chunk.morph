{
    "name": "hello",
    "kind": "chunk",
    "source": {
        "repo": "hello",
        "ref": "master"
    },
    "configure-commands": [
        "./configure --prefix=/usr"
    ],
    "build-commands": [
        "make"
    ],
    "install-commands": [
        "make DESTDIR=\"$DESTDIR\" install"
    ]
}
