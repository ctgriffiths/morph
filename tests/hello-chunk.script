#!/bin/sh
#
# Test building a chunk for GNU Hello.
#
# Copyright (C) 2011  Codethink Limited
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

set -e

tar -C "$DATADIR" -xf tests/hello-chunk.tar.gz
./morph build "$DATADIR/hello/hello.morph" \
    --chunk-repo="file://$DATADIR/hello" --chunk-ref=master \
    --cachedir="$DATADIR"
tar -tf "$DATADIR/"*.chunk | LC_ALL=C sort
