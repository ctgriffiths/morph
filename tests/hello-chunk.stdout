.chunk.hello:
./
baserock/
baserock/hello.meta
usr/
usr/local/
usr/local/bin/
usr/local/bin/greeter

.chunk.hello-doc:
./
baserock/
baserock/hello-doc.meta
usr/
usr/local/
usr/local/man/
usr/local/man/man1/
usr/local/man/man1/hello.1

